receiver/snmpreceiver/config.go

```diff
@@ -448,18 +455,28 @@ func validateColumnOID(metricName string, columnOID ColumnOID, cfg *Config) erro
 
 	// Check that any ResourceAttributes have a valid value
 	if len(columnOID.ResourceAttributes) > 0 {
-		hasIndexedIdentifier = true
 		for _, name := range columnOID.ResourceAttributes {
-			_, ok := cfg.ResourceAttributes[name]
+			resourceAttribute, ok := cfg.ResourceAttributes[name]
 			if !ok {
 				combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgColumnResourceAttributeBadName, metricName, name))
+				continue
+			} 
+
+			if resourceAttribute.OID != "" || resourceAttribute.IndexedValuePrefix != "" { hasIndexedIdentifier = true }
```

Comment:
Does it make sense to move this check until after we've validated that only one of oid, scaler_oid, indexed_prefix_value is set? The flag that it sets is used immediately after this loop, and it's generally nice to keep related code adjacent unless there is a reason not to.

Reply:
Moved the assignment into those validation checks since they are duplicated otherwise.